vercors/src/main/scala/vct/col/rewrite/RewriteArrayRef.scala

```diff
@@ -359,7 +359,7 @@ class RewriteArrayRef(source: ProgramUnit) extends AbstractRewriter(source) {
     var rowJ: ASTNode = create.expression(StandardOperator.Subscript, matrix, quantJ0)
 
     if(seqInfo0.isCell) {
-      conditions += create.starall(quantGuard, create.expression(StandardOperator.Perm, row), quantDecls: _*)
+      conditions += create.starall(quantGuard, create.expression(StandardOperator.Perm, row, create.reserved_name(ASTReserved.ReadPerm)), quantDecls: _*)
```

Comment:
What is this change doing? I can see you are asking read permissions specifically, but what was it doing before when the read permissions were not specified?